components/list/list-item-menu-item-mixin.js

```diff
@@ -0,0 +1,154 @@
+import '../colors/colors.js';
+import { css, html } from 'lit';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ListItemMixin } from './list-item-mixin.js';
+import { EventSubscriberController } from '../../controllers/subscriber/subscriberControllers.js';
+
+const interactiveElements = {
+	// 'a' only if an href is present
+	'button': true,
+	'input': true,
+	'select': true,
+	'textarea': true,
+	'd2l-tooltip-help': true,
+	'd2l-button': true
+};
+
+const interactiveRoles = {
+	'button': true,
+	'checkbox': true,
+	'combobox': true,
+	'link': true,
+	'listbox': true,
+	'menuitem': true,
+	'menuitemcheckbox': true,
+	'menuitemradio': true,
+	'option': true,
+	'radio': true,
+	'slider': true,
+	'spinbutton': true,
+	'switch': true,
+	'tab:': true,
+	'textbox': true,
+	'treeitem': true
+};
+
+function getIsInteractiveChildClicked(e, linkElem) {
+	const composedPath = e.composedPath();
+	for (let i = 0; i < composedPath.length; i++) {
+		const elem = composedPath[i];
+		if (!elem.getAttribute) continue;
+
+		if (elem === linkElem) {
+			return false;
+		}
+
+		const nodeName = elem.nodeName.toLowerCase();
+		if (interactiveElements[nodeName] || (nodeName === 'a' && elem.hasAttribute('href'))) {
+			return true;
+		}
+
+		const role = (elem.getAttribute('role') || '');
+		if (interactiveRoles[role]) {
+			return true;
+		}
+	}
+	return false;
+}
+
+export const ListItemMenuItemMixin = superclass => class extends ListItemMixin(superclass) {
```

Comment:
We're definitely blurring the lines between menu and list. Maybe that's ok. Alternative might be to make it more specific to navigation. ex. `ListItemNavItem(Mixin)`.